source/Applications/Advanced/TransformViaArucoMarker/TransformViaArucoMarker.cpp

```diff
@@ -0,0 +1,315 @@
+
+        auto u = xAxis / cv::norm(xAxis, cv::NORM_L2);
+        auto v = yAxis / cv::norm(yAxis, cv::NORM_L2);
+        auto normal = u.cross(v);
+        auto unitNormal = normal / cv::norm(normal, cv::NORM_L2);
+
+        auto rotationMatrix = cv::Mat(3, 3, CV_64F);
+        for(int i = 0; i < 3; ++i)
+        {
+            rotationMatrix.at<double>(i, 0) = u[i];
+            rotationMatrix.at<double>(i, 1) = v[i];
+            rotationMatrix.at<double>(i, 2) = unitNormal[i];
+        }
+
+        auto transform = transformationMatrix(rotationMatrix, origin);
+
+        std::cout << "Displaying marker's pose" << std::endl;
+        displayMarkerPose(pointCloud, corners2D, center2D);
+
+        return transform;
+    }
+
+    Zivid::Matrix4x4 invertAffineTransformation(const Zivid::Matrix4x4 &transform)
+    {
+        auto rotation = cv::Mat(3, 3, CV_32F);
+        for(int i = 0; i < 3; ++i)
+        {
+            for(int j = 0; j < 3; ++j)
+            {
+                rotation.at<float>(i, j) = transform(i, j);
+            }
+        }
+
+        auto translation = cv::Mat(3, 1, CV_32F);
+        for(int i = 0; i < 3; ++i)
+        {
+            translation.at<float>(i, 0) = transform(i, 3);
+        }
+
+        cv::Mat newRotation = rotation.inv();
+        cv::Mat newTranslation = -newRotation * translation;
+
+        auto cvInverted = cv::Mat(4, 4, CV_32F, 0.0);
+        for(int i = 0; i < 3; ++i)
+        {
+            for(int j = 0; j < 3; ++j)
+            {
+                cvInverted.at<float>(i, j) = newRotation.at<float>(i, j);
+            }
+        }
+        for(int i = 0; i < 3; ++i)
+        {
+            cvInverted.at<float>(i, 3) = newTranslation.at<float>(i, 0);
+        }
+        cvInverted.at<float>(3, 3) = 1.0;
+
+        auto *ptr = &cvInverted.at<float>(0, 0);
+        auto invertedTransform = Zivid::Matrix4x4(ptr, ptr + 16);
+
+        return invertedTransform;
+    }
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        const auto dataFile = std::string(ZIVID_SAMPLE_DATA_DIR) + "/ArucoSample.zdf";
+        std::cout << "Reading ZDF frame from file: " << dataFile << std::endl;
+        const Zivid::Frame frame = Zivid::Frame(dataFile);
+        auto pointCloud = frame.pointCloud();
+
+        std::cout << "Converting to OpenCV BGR image format" << std::endl;
+        cv::Mat image = pointCloudToCvBGR(pointCloud);
+
+        std::cout << "Detecting aruco markers in 2D image" << std::endl;
+        auto markers = aruconano::MarkerDetector::detect(image);
+
+        std::cout << "Drawing marker corners" << std::endl;
+        for(const auto &marker : markers)
+        {
+            marker.draw(image);
+        }
+        std::cout << "Displaying detected markers" << std::endl;
+        cv::imshow("Detected aruco markers", image);
+        cv::waitKey(0);
+
+        std::cout << "Detecting aruco marker poses" << std::endl;
```

Comment:
'ArUco'

Reply:
i agree, its not necessary to have all those comments. done